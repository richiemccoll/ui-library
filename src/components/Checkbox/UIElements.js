import styled, { css } from "styled-components";

export const Container = styled.div`
    display: flex;
    backface-visibility: hidden;
    outline: 0;
    vertical-align: baseline;
    font-style: normal;
    min-height: 17px;
    font-size: 1rem;
    line-height: 17px;
    min-width: 17px;
`;

export const Label = styled.label`
  color: black;
  padding: .5em;
  vertical-align: middle;

  ${props =>
    props.disabled &&
    css`
    color: #ddd;
  `}
`;

export const Input = styled.input`
    -moz-appearence: none;
    -webkit-appearance: none;
    height: 25px;
    width: 25px;
    border: 1px solid #ddd;
    transition: all 0.25s;
    border-radius: 2px;

    &:checked {
      background-color: #2196F3;
      outline: 0;

      ${props =>
      props.type === "radio" &&
      css`
        background-color: #2196F3;
        border-radius: 15px;
    `
    }
}
    ${props =>
      props.type === "radio" &&
      css`
        border-radius: 15px;
  `}
`;
