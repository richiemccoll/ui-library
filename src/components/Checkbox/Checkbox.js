import React, { Component } from "react";
import PropTypes from "prop-types";
import { Container, Input, Label } from "./UIElements";

class Checkbox extends Component {
  constructor(props) {
    super(props);
    this.state = {
      checked: false
    };
    this.toggleCheckbox = this.toggleCheckbox.bind(this);
    this.isRadio = this.isRadio.bind(this);
  }

  toggleCheckbox() {
    this.setState({ checked: !this.state.checked });
  }

  isRadio(type) {
    return type === "radio";
  }

  render() {
    const { name, type, disabled = false } = this.props;

    return (
      <Item
        inputName={this.isRadio(type) ? "Radio" : name}
        name={name}
        type={type}
        disabled={disabled}
        checked={this.state.checked}
        handleChange={this.toggleCheckbox}
      />
    );
  }
}

export default Checkbox;

Checkbox.PropTypes = {
  name: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  disabled: PropTypes.bool
};

const Item = ({ inputName, name, type, disabled, checked, handleChange }) => {
  return (
    <Container>
      <Input
        name={inputName}
        id={name}
        type={type}
        disabled={disabled}
        checked={checked}
        onChange={handleChange}
      />
      <Label disabled={disabled} htmlFor={name}>{name}</Label>
    </Container>
  );
};
