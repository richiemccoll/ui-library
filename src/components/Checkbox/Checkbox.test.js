import React from "react";
import Checkbox from "./Checkbox";
import renderer from "react-test-renderer";

describe("Checkbox Component", () => {
  it("should render a checkbox that is not disabled", () => {
    const component = renderer.create(<Checkbox name="test" type="checkbox" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render a checkbox that is disabled", () => {
    const component = renderer.create(
      <Checkbox name="test" type="checkbox" disabled="true" />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should update the checked property on click", () => {
    const component = renderer.create(<Checkbox name="test" type="radio" />);
    // initial render
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree.children[0].props.checked).toBe(false);

    tree.children[0].props.onChange();
    //re-render and verify
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    expect(tree.children[0].props.checked).toBe(true);
  });
});
