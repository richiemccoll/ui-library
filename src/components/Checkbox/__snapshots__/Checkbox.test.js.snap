// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Component should render a checkbox that is disabled 1`] = `
<div
  className="sc-bdVaJa fkJLfL"
>
  <input
    checked={false}
    className="sc-htpNat dOJGmE"
    disabled="true"
    id="test"
    name="test"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="sc-bwzfXH gBgBSn"
    disabled="true"
    htmlFor="test"
  >
    test
  </label>
</div>
`;

exports[`Checkbox Component should render a checkbox that is not disabled 1`] = `
<div
  className="sc-bdVaJa fkJLfL"
>
  <input
    checked={false}
    className="sc-htpNat dOJGmE"
    disabled={false}
    id="test"
    name="test"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="sc-bwzfXH ctswpo"
    disabled={false}
    htmlFor="test"
  >
    test
  </label>
</div>
`;

exports[`Checkbox Component should update the checked property on click 1`] = `
<div
  className="sc-bdVaJa fkJLfL"
>
  <input
    checked={false}
    className="sc-htpNat dgdDeR"
    disabled={false}
    id="test"
    name="Radio"
    onChange={[Function]}
    type="radio"
  />
  <label
    className="sc-bwzfXH ctswpo"
    disabled={false}
    htmlFor="test"
  >
    test
  </label>
</div>
`;

exports[`Checkbox Component should update the checked property on click 2`] = `
<div
  className="sc-bdVaJa fkJLfL"
>
  <input
    checked={true}
    className="sc-htpNat dgdDeR"
    disabled={false}
    id="test"
    name="Radio"
    onChange={[Function]}
    type="radio"
  />
  <label
    className="sc-bwzfXH ctswpo"
    disabled={false}
    htmlFor="test"
  >
    test
  </label>
</div>
`;
